packages/react-native/Libraries/Blob/URLSearchParams.js

```diff
@@ -11,60 +11,129 @@
 // Small subset from whatwg-url: https://github.com/jsdom/whatwg-url/tree/master/src
 // The reference code bloat comes from Unicode issues with URLs, so those won't work here.
 export class URLSearchParams {
-  _searchParams: Array<[string, string]> = [];
+  _searchParams: Map<string, string[]> = new Map();
 
-  constructor(params?: Record<string, string>) {
-    if (typeof params === 'object') {
-      Object.keys(params).forEach(key => this.append(key, params[key]));
+  constructor(params?: Record<string, string> | string | [string, string][]) {
+    // URLSearchParams("key1=value1&key2=value2");
+    if (typeof params === 'string') {
+      params
+        .replace(/^\?/, '')
+        .split('&')
+        .forEach(pair => {
+          if (!pair) {
+            return;
+          }
+          const [key, value] = pair.split('=').map(decodeURIComponent);
+          this.append(key, value);
+        });
+    }
+
+    //URLSearchParams([["key1", "value1"], ["key2", "value2"]]);
+    else if (Array.isArray(params)) {
+      params.forEach(([key, value]) => this.append(key, value));
+    }
+    //URLSearchParams({ key1: "value1", key2: "value2" });
+    else if (typeof params === 'object') {
+      Object.entries(params).forEach(([key, value]) => this.append(key, value));
     }
   }
 
   append(key: string, value: string): void {
-    this._searchParams.push([key, value]);
+    if (!this._searchParams.has(key)) {
+      this._searchParams.set(key, [value]); // Initialize with an array if key is missing
+    } else {
+      this._searchParams.get(key)?.push(value); // Else push the value to the array
+    }
+  }
+
+  delete(name: string): void {
+    this._searchParams.delete(name);
+  }
+
+  get(name: string): string | null {
+    const values = this._searchParams.get(name);
+    return values ? values[0] : null;
+  }
+
+  getAll(name: string): string[] {
+    return this._searchParams.get(name) ?? [];
   }
 
-  delete(name: string): empty {
-    throw new Error('URLSearchParams.delete is not implemented');
+  has(name: string): boolean {
+    return this._searchParams.has(name);
   }
 
-  get(name: string): empty {
-    throw new Error('URLSearchParams.get is not implemented');
+  set(name: string, value: string): void {
+    this._searchParams.set(name, [value]);
   }
 
-  getAll(name: string): empty {
-    throw new Error('URLSearchParams.getAll is not implemented');
+  keys(name: string): Iterator<string> {
+    return this._searchParams.keys();
   }
 
-  has(name: string): empty {
-    throw new Error('URLSearchParams.has is not implemented');
+  values(name: string): Iterator<string> {
+    function* generateValues(params: Map<string, string[]>): Iterator<string> {
+      for (const valueArray of params.values()) {
+        for (const value of valueArray) {
+          yield value;
+        }
+      }
+    }
+    return generateValues(this._searchParams);
+  }
+
+  entries(name: string): Iterator<[string, string]> {
```

Comment:
(Pre-existing) These functions don't have a name arg.
```suggestion
  values(): Iterator<string> {
    function* generateValues(params: Map<string, string[]>): Iterator<string> {
      for (const valueArray of params.values()) {
        for (const value of valueArray) {
          yield value;
        }
      }
    }
    return generateValues(this._searchParams);
  }

  entries(): Iterator<[string, string]> {
```

Reply:
Removed args